### PR TITLE
Ses 105 feature layout and implement 404 page v 10

### PR DESCRIPTION
**Pull request by alejandrocabriales**
# Ticket
https://trello.com/c/lBsWwHWo/105-feature-layout-and-implement-404-page-v-10

# Description
404 not found page

# Actions
Add component not found and Page not found
Add the necessary asserts
Add dark mode and asserts
Add props light

### CODE REVIEW

File: src/stories/components/404/not-found-component-404.tsx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  const isMobile = useMediaQuery(lightTheme.breakpoints.between('table_375', 'table_834'));
+  const isLight = useThemeContext().themeMode === 'light';
+  const handleOnclick = () => {
+    router.push('/');
+  };
+  return (
+    <Wrapper>
+      <ImageContainer>
+        <Image
+          src={isLight && (phoneLess || isMobile) ? BackgroundMobile404 : !isLight && (phoneLess || isMobile) ? BackgroundMobile404Dark : isLight && !(phoneLess || isMobile) ? Background404 : Background404Dark}
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:29]: As this line is too big, maybe could make sense to format it this way (so it is more readable):

```js
isLight ? (
  // conditions for light theme
) : (
  // conditions for dark theme
)
```

File: src/stories/components/404/not-found-component-404.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  [lightTheme.breakpoints.between('desktop_1194', 'desktop_1280')]: {
+    marginTop: '132px',
+  },
+  [lightTheme.breakpoints.between('desktop_1194', 'desktop_1280')]: {
+    marginTop: '132px',
+  },
+  [lightTheme.breakpoints.between('desktop_1280', 'desktop_1440')]: {
+    marginTop: '132px',
+  },
+  [lightTheme.breakpoints.between('desktop_1440', 'desktop_1920')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:86]: All the medias have the same property, is it possible to use instead:

```css
[lightTheme.breakpoints.between('table_834', 'desktop_1920')]: {
    marginTop: '132px',
 },
```

File: src/stories/components/404/not-found-component-404.tsx [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  height: '164px',
+  marginBottom: '64px',
+  marginLeft: '32px',
+  marginRight: '32px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1194')]: {
+    marginBottom: '160px',
+    width: '580px',
+    height: '340px',
+  },
+  [lightTheme.breakpoints.between('desktop_1194', 'desktop_1280')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:163]: All the media have the same properties, is it possible to use instead:

```css
[lightTheme.breakpoints.up('desktop_1194')]: {
    marginBottom: '37.26px',
    width: '675px',
    height: '397.74px',
  },
```

File: src/stories/components/404/not-found-component-404.tsx [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+const ContainerText = styled.div({
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  paddingRight: '32px',
+  paddingLeft: '32px',
+  maxWidth: '343px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1194')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:193]: All the media have the same properties, is it possible to use instead:

```css
[lightTheme.breakpoints.up('table_834')]: {
    maxWidth: '100%',
  },
```

File: src/stories/components/404/not-found-component-404.tsx [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+  fontFamily: 'FT Base, sans-serif',
+  fontStyle: 'normal',
+  fontWeight: 700,
+  fontSize: '36px',
+  lineHeight: '43px',
+  letterSpacing: '0.4px',
+  color: isLight ? '#787A9B' : '#D2D4EF',
+  textAlign: 'center',
+  marginBottom: '24px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1194')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:220]: Same recommendation about the same media properties than above :)

File: src/stories/components/404/not-found-component-404.tsx [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+
+const TextDescription = styled(Typography)<{ isLight?: boolean }>(({ isLight }) => ({
+  fontFamily: 'FT Base,san-serif',
+  fontStyle: 'normal',
+  fontWeight: 500,
+  fontSize: '20px',
+  lineHeight: '24px',
+  textAlign: 'center',
+  letterSpacing: '0.4px',
+  color: isLight ? '#ADAFD4' : '#ADAFD4',
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:260]: This `TextDesciption` has the same color for light and dark themes, is it ok? If it is maybe it is a good idea just to use `color: #ADAFD4`

File: src/stories/components/404/not-found-component-404.tsx [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+  marginBottom: '64px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1194')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:262]: Same recommendation about media

File: src/stories/components/404/not-found-component-404.tsx [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+    lineHeight: '29px',
+    marginBottom: '40px',
+  }
+}));
+
+const ContainerButton = styled.div({
+  marginBottom: '83px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1194')]: {
+  },
+  [lightTheme.breakpoints.between('desktop_1194', 'desktop_1280')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:293]: Same recommendation about media. Please, note that the previous media is empty

**Review by YasielCabrera — COMMENTED** [review]
Remember to compress all the images to save some KB

File: src/stories/components/404/not-found-component-404.tsx (hunk 1 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are right, already done

File: src/stories/components/404/not-found-component-404.tsx (hunk 2 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 3 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 4 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 5 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 6 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: My mistake, already fixed

File: src/stories/components/404/not-found-component-404.tsx (hunk 2 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are right

File: src/stories/components/404/not-found-component-404.tsx (hunk 3 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 4 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx (hunk 5 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: thanks

File: src/stories/components/404/not-found-component-404.tsx (hunk 6 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: fixed

File: src/stories/components/404/not-found-component-404.tsx (hunk 8 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: Fixed

File: src/stories/components/404/not-found-component-404.tsx (hunk 7 above)
Comment (alejandrocabriales) [reply src/stories/components/404/not-found-component-404.tsx]: you are rights, i will fix it

File: src/stories/components/404/not-found-component-404.tsx [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+      </ImageContainer>
+    </Wrapper >);
+};
+
+const Wrapper = styled.div({
+  width: '100%',
+  height: '100%',
+  marginTop: '132px',
+  marginBottom: '128px',
+  [lightTheme.breakpoints.between('table_834', 'desktop_1440')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:79]: I just noted that you're already using `marginTop: '132px'` as a default value, so this media query is no longer needed

File: src/stories/components/404/not-found-component-404.tsx [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+  [lightTheme.breakpoints.up('table_834')]: {
+    fontSize: '24px',
+    lineHeight: '29px',
+    marginBottom: '40px',
+  },
+});
+
+const ContainerButton = styled.div({
+  marginBottom: '83px',
+  [lightTheme.breakpoints.up('table_834')]: {
```
Comment (YasielCabrera) [inline src/stories/components/404/not-found-component-404.tsx:203]: Did you forget to remove this empty media query?

**Review by YasielCabrera — APPROVED** [review]
LGTM

**Pull request merged**